### PR TITLE
COTR-15-Movement

### PR DESCRIPTION
**Pull request by Elijuszek**
Created player scene and implemented basic movement. Commented the code in-depth.

### CODE REVIEW

**Comment by Karolinskis** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/nekompiliuoja/champions-of-the-realm) found an issue: A closing curly bracket must not be preceded by a blank line.

It's currently on:
[Scenes\Actors\Player\Player.cs:111](https://github.com/Nekompiliuoja/Champions-of-the-Realm/blob/784458fbb8502284f7285d06b92579e1b6bf11aa/Scenes/Actors/Player/Player.cs#L111)
Commit 784458fbb8502284f7285d06b92579e1b6bf11aa

**Comment by Karolinskis** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/nekompiliuoja/champions-of-the-realm) found an issue: The code must not contain multiple blank lines in a row.

It's currently on:
[Scenes\Actors\Player\Player.cs:31](https://github.com/Nekompiliuoja/Champions-of-the-Realm/blob/784458fbb8502284f7285d06b92579e1b6bf11aa/Scenes/Actors/Player/Player.cs#L31)
Commit 784458fbb8502284f7285d06b92579e1b6bf11aa

File: Scenes/Actors/Actor.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	private Vector2 knockBack = Vector2.Zero;
+
+	public override void _Ready()
+	{
+		base._Ready();
+		Stats = GetNode<Stats>("Stats");
+        team = GetNode<Team>("Team");
+        collisionShape = GetNode<CollisionShape2D>("CollisionShape2D");
+
+        // TODO: Can't implement blood particales, because blood scene is missing
```
Comment (Karolinskis) [inline Scenes/Actors/Actor.cs:23]: Create an issue for this TODO

File: Scenes/Actors/Player/Player.cs [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    public delegate void PlayerHealthChangedEventHandler(float newHealth);
+    [Signal]
+    public delegate void PLayerGoldChangedEventHandler(int newGold, int oldGold);
+    [Signal]
+    public delegate void DiedEventHandler();
+
+    [Export] float swingDuration = 0.5f; // TODO swing stab pierce hit
+    [Export] float reloadDuration = 1f;
+
+    //TODO: lacking Joystick scene implementation
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:17]: Create an issue for this TODO

File: Scenes/Actors/Player/Player.cs [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    //private Joystick movementJoystick;
+    //private Joystick attackJoystick;
+
+    private AnimationPlayer animationPlayer;
+    private RemoteTransform2D cameraTransform;
+    private AudioStreamPlayer coinsSound;
+    private Vector2 movementDirection = Vector2.Zero;
+    private Vector2 attackDirection = Vector2.Zero;
+
+    // TODO: lacking damagePopup scene implementation
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:27]: Create an issue for this TODO

File: Scenes/Actors/Player/Player.cs [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+    private Globals globals;
+    public WeaponsManager WeaponsManager { get; set; }
+    public override void _Ready()
+	{
+        base._Ready();
+        animationPlayer = GetNode<AnimationPlayer>("AnimationPlayer");
+        WeaponsManager = GetNode<WeaponsManager>("WeaponsManager");
+
+        // TODO: lacking weaponsManager, GUI and Joystick implementation
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:39]: Create an issue for this TODO

File: Scenes/Actors/Player/Player.cs [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+                PlayWalking();
+            }
+            else
+            {
+                PlayIdle();
+            }
+        }
+        Velocity = movementDirection * Stats.Speed;
+
+        // TODO: lacking Joystick scene implementation
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:66]: Create an issue for this TODO

File: Scenes/Actors/Player/Player.cs [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+
+    /// <summary>
+    /// Method for changing weapon
+    /// </summary>
+    public void ChangeWeapon()
+    {
+        //WeaponsManager.ChangeWeapon();
+    }
+
+    // TODO: lacks WeaponsManager scene implementation and animations
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:149]: Create an issue for this TODO

File: Scenes/Actors/Player/Player.cs [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+    /// </summary>
+    /// <param name="removeAmount">Gold amount to remove</param>
+    public void RemoveGold(int removeAmount)
+    {
+        int oldGold = Stats.Gold;
+        Stats.Gold -= removeAmount;
+        EmitSignal(nameof(PLayerGoldChanged), oldGold, Stats.Gold);
+    }
+
+    // TODO: lacks WeaponsManager scene implementation
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:132]: Create an issue for this TODO

File: Scenes/Actors/Actor.cs (hunk 1 above)
Comment (Elijuszek) [reply Scenes/Actors/Actor.cs:23]: Added task COTR-29 for this issue.

File: Scenes/Actors/Player/Player.cs (hunk 2 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs:17]: Added task COTR-30 for this issue.

File: Scenes/Actors/Player/Player.cs (hunk 3 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs:27]: Added task COTR-31 for this issue.

File: Scenes/Actors/Player/Player.cs (hunk 4 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs:39]: Added task COTR-32 for this issue.

File: Scenes/Actors/Player/Player.cs (hunk 5 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs:66]: Added task COTR-30 for this issue.

File: Scenes/Actors/Player/Player.cs (hunk 6 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs:149]: Added task COTR-32 for this issue.

File: Scenes/Actors/Player/Player.cs (hunk 7 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs:132]: Added task COTR-32 for this issue.

File: Scenes/Actors/Actor.cs [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        MoveAndSlide();
+	}
+
+    /// <summary>
+    /// Method for handeling received damage
+    /// Should only be fully implemented in deriveted class
+    /// </summary>
+    /// <param name="baseDamage">Amount of damage received</param>
+    /// <param name="impactPosition">Position for spawning blood particales</param>
+    public virtual void HandleHit(float baseDamage, Vector2 impactPosition) { GD.PrintErr("Calling HandleHit from Actor class!"); }
```
Comment (Karolinskis) [inline Scenes/Actors/Actor.cs:40]: Might want to make this shorter

**Review by Karolinskis — APPROVED** [review]
LGTM!

**Pull request merged**